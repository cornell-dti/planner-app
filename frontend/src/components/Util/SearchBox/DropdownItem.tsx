import React, { ReactElement } from 'react';
import { FuseItem } from './types';


type Props<T extends FuseItem> = {
  readonly item: T;
  readonly className: string;
  readonly onSelect: (selected: T) => void;
}

export default <T extends FuseItem>({ item, className, onSelect }: Props<T>): ReactElement => (
  <button type="button" className={className} title={'Add class'} onClick={() => onSelect(item)} tabIndex={0}>
    {item.value}
  </button>
);
