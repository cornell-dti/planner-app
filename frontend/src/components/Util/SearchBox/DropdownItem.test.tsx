import React from 'react';
import renderer from 'react-test-renderer';
import DropdownItem from './DropdownItem';

it('DropdownItem matches snapshot.', () => {
  const tree = renderer
    .create(
      <DropdownItem item={{ key: 42, value: 'foo-bar' }} className="test" onSelect={() => {}} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
