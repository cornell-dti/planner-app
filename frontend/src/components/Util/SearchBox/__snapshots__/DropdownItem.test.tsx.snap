// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem matches snapshot. 1`] = `
<button
  className="test"
  onClick={[Function]}
  tabIndex={0}
  title="Add class"
  type="button"
>
  foo-bar
</button>
`;
